### DISCUSSION TITLE
DataTable and Dates

### DISCUSSION THREAD

**Discussion by wtrudell360** [carbon-design-system/carbon#10149] [Help] [answered]
Is there a way to format a Date within the datatable and have it sort correctly as a Date? I can format it and have it display the way I need but it of course sorts a text not as a Date.

**Comment by tay1orjones** [comment, marked as answer]
Hi @wtrudell360, custom sorting is supported through the `sortRow` prop available on `DataTable`. Through it you provide your own function to control sorting beyond the default text sort. Full info is available in the [`DataTable` Custom Sorting docs](https://react.carbondesignsystem.com/?path=/docs/components-datatable-sorting--usage#custom-sorting) in storybook

**Reply by wtrudell360** [reply]
Thanks

**Comment by vmukhachev** [comment]
@tay1orjones , could you please explain how to use it for dates sorting? custom "sortRow" function receives the formatted value and it is not possible to get the raw values without parsing them.

**Reply by tay1orjones** [reply]
@vmukhachev Yeah, you'll probably need to parse them for thorough comparison. You might find something like `moment.js` helpful in this scenario, but generally speaking it should be performant to parse the date as part of the comparison between values.

Worth noting that depending on the format you can rely on the strings being compared.

```js
if ("2022-05-31" < "2012-06-01") {
    return -1;
}
```
